driver-core/src/main/com/mongodb/internal/connection/AsynchronousChannelStream.java

```diff
@@ -107,19 +109,19 @@ public void failed(final Throwable t) {
     }
 
     @Override
-    public void readAsync(final int numBytes, final AsyncCompletionHandler<ByteBuf> handler) {
-        readAsync(numBytes, 0, handler);
+    public void readAsync(final int numBytes, final OperationContext operationContext, final AsyncCompletionHandler<ByteBuf> handler) {
+        readAsync(numBytes, 0, operationContext, handler);
     }
 
-    private void readAsync(final int numBytes, final int additionalTimeout, final AsyncCompletionHandler<ByteBuf> handler) {
+    private void readAsync(final int numBytes, final int additionalTimeout, final OperationContext operationContext,
+            final AsyncCompletionHandler<ByteBuf> handler) {
         ByteBuf buffer = bufferProvider.getBuffer(numBytes);
 
-        int timeout = settings.getReadTimeout(MILLISECONDS);
+        long timeout = operationContext.getTimeoutContext().getReadTimeoutMS();
```

Comment:
Additional timeout will be removed as part of JAVA-4063 (commented where in the DefaultServerMonitor).